picard/ui/formattedtextdelegate.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2025 Philipp Wolfer
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import math
+
+from PyQt6 import (
+    QtCore,
+    QtGui,
+    QtWidgets,
+)
+
+
+class FormattedTextDelegate(QtWidgets.QStyledItemDelegate):
+    def __init__(self, parent=None, markup_format='html'):
+        """
+        A QStyledItemDelegate that renders formatted text in a QTreeWidget.
+
+        :param markup_format: Specify the format of the markup, either "html" or "markdown".
+        """
+        super().__init__(parent)
+        self.markup_format = markup_format
+
+    def paint(self, painter, option, index):
+        # Initialize the style option
+        self.initStyleOption(option, index)
+
+        # Draw the background
+        if option.state & QtWidgets.QStyle.StateFlag.State_Selected:
+            painter.fillRect(option.rect, option.palette.highlight())
+        else:
+            painter.fillRect(option.rect, option.palette.base())
```

Comment:
Just a minor thing, reduce code redundancy a bit:

```python
        if option.state & QtWidgets.QStyle.StateFlag.State_Selected:
            fill_color = option.palette.highlight()
        else:
            fill_color = option.palette.base()
        painter.fillRect(option.rect, fill_color)
```

And group with `text_color` below

Reply:
Done, updated the code in the latest commit